scripts/_draw_info_display.py

```diff
@@ -0,0 +1,226 @@
+import math
+import os
+
+config = configparser.ConfigParser()
+
+config.read(os.path.join(utils.get_script_dir(), 'Modules', 'infodisplay.ini'))
+
+debug = config['DEBUG'].getboolean('Debug')
+
+frame_count = config['INFO DISPLAY'].getboolean('Frame Count')
+
+lap_splits = config['INFO DISPLAY'].getboolean('Lap Splits')
+
+speed = config['INFO DISPLAY'].getboolean('Speed')
+
+internal_velocity = config['INFO DISPLAY'
+                    ].getboolean('Internal Velocity (X, Y, Z)')
+internal_velocity_xyz = config['INFO DISPLAY'
+                        ].getboolean('Internal Velocity (XYZ)')
+
+external_velocity = config['INFO DISPLAY'
+                    ].getboolean('External Velocity (X, Y, Z)')
+external_velocity_xyz = config['INFO DISPLAY'
+                        ].getboolean('External Velocity (XYZ)')
+
+moving_road_velocity = config['INFO DISPLAY'
+                       ].getboolean('Moving Road Velocity (X, Y, Z)')
+moving_road_velocity_xyz = config['INFO DISPLAY'
+                           ].getboolean('Moving Road Velocity (XYZ)')
+
+moving_water_velocity = config['INFO DISPLAY'
+                        ].getboolean('Moving Water Velocity (X, Y, Z)')
+moving_water_velocity_xyz = config['INFO DISPLAY'
+                            ].getboolean('Moving Water Velocity (XYZ)')
+
+charges_and_boosts = config['INFO DISPLAY'
+                     ].getboolean('Charges and Boosts')
+checkpoints_and_completion = config['INFO DISPLAY'
+                             ].getboolean('Checkpoints and Completion')
+miscellaneous = config['INFO DISPLAY'
+                ].getboolean('Miscellaneous')
+
+surface_properties = config['INFO DISPLAY'
+                     ].getboolean('Surface Properties')
+
+position = config['INFO DISPLAY'].getboolean('Position')
+
+stick = config['INFO DISPLAY'].getboolean('Stick')
+
+color = int(config['INFO DISPLAY']['Text Color'], 16)
+
+digits = config['INFO DISPLAY'].getint('Digits (to round to)')
+
+# draw information to the screen
+
+
+def create_infodisplay():
+    text = ""
+
+    if debug:
+        # test values here
+        text += f"{utils.get_game_id()}\n\n"
+    
+    if frame_count:
+        text += f"Frame: {core.get_frame_of_input()}\n\n"
+
+    if lap_splits:
+        # The actual max lap address does not update when crossing the finish line
+        # for the final time to finish the race. However, for whatever reason,
+        # race completion does. We use the "max" version to prevent lap times
+        # from disappearing when crossing the line backwards.
+        player_max_lap = math.floor(
+            classes.RaceInfoPlayer.race_completion_max())
+        lap_count = classes.RaceDataSettings.lap_count()
+
+        if player_max_lap >= 2 and lap_count > 1:
+            for lap in range(1, player_max_lap):
+                text += "Lap {:d}: {:02d}:{:012.9f}\n".format(
+                    lap, core.updateExactFinish(lap, 0)[0],
+                    core.updateExactFinish(lap, 0)[1])
```

Comment:
Just call this function once. You can do that within the `format()` call using argument unpacking.

    if player_max_lap >= 2 and lap_count > 1:
        for lap in range(1, player_max_lap):
            text += "Lap {:d}: {:02d}:{:012.9f}\n".format(
                lap, *core.updateExactFinish(lap, 0))

Reply:
Also, shouldn't we be factoring in the "digits" argument?